mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/resourceprovider/ResourceClusterProviderUpgradeRequest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.resourcecluster.resourceprovider;
+
+import io.mantisrx.master.resourcecluster.proto.MantisResourceClusterEnvType;
+import io.mantisrx.master.resourcecluster.proto.MantisResourceClusterSpec;
+import io.mantisrx.master.resourcecluster.proto.UpgradeClusterContainersRequest;
+import io.mantisrx.server.master.resourcecluster.ClusterID;
+import javax.annotation.Nullable;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder(toBuilder = true)
+public class ResourceClusterProviderUpgradeRequest {
+    ClusterID clusterId;
+
+    String region;
+
+    String optionalImageId;
+
+    String optionalSkuId;
+
+    MantisResourceClusterEnvType optionalEnvType;
+
+    int optionalBatchMaxSize;
```

Comment:
what is this batchMaxSize for?

Reply:
Per upgrade batch settings where each batch in upgrade should not have more than x agents involved.